File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/05. CLIESCRITORIO/cliente_escritorio_soap_dotnet_conuni_gr06/src/Modelo/ServicioAutenticacion.cs
namespace Ec.Edu.Monster.Modelo;

public class ServicioAutenticacion
{
    private readonly ClienteSoap cliente = new();

    public bool Autenticar(string usuario, string contrasena)
    {
        return cliente.Ejecutar(servicio => servicio.IniciarSesion(usuario, contrasena));
    }
}

File: SIN BDD/SOAP_JAVA_CONUNI_GR06/06. CLIWEB/cliente_web_soap_java_conuni_gr06/src/java/ec/edu/monster/modelo/ServicioAutenticacion.java
package ec.edu.monster.modelo;

import java.util.LinkedHashMap;
import java.util.Map;

public class ServicioAutenticacion {

    private final ClienteSoap clienteSoap = new ClienteSoap();

    public boolean iniciarSesion(String usuario, String contrasena) throws Exception {
        Map<String, String> parametros = new LinkedHashMap<>();
        parametros.put("usuario", usuario);
        parametros.put("contrasena", contrasena);
        String respuesta = clienteSoap.invocar("iniciarSesion", parametros);
        return Boolean.parseBoolean(respuesta);
    }
}
